app/javascript/Components/course_summaries_table.jsx

```diff
@@ -1,121 +1,160 @@
-      headerClassName: "rt-hidden",
-      resizable: false,
-      width: 0,
-    },
-    {
-      Header: I18n.t("activerecord.attributes.user.user_name"),
-      accessor: "user_name",
-      filterable: true,
-    },
-    {
-      Header: I18n.t("activerecord.attributes.user.first_name"),
-      accessor: "first_name",
-      filterable: true,
-    },
-    {
-      Header: I18n.t("activerecord.attributes.user.last_name"),
-      accessor: "last_name",
-      filterable: true,
-    },
-  ];
+  nameColumns = () => {
+    const columnHelper = this.columnHelper;
+    return [
+      columnHelper.accessor("hidden", {
+        id: "hidden",
+        filterFn: (row, columnId, filterValue) => {
+          // Show all rows if filter true, else only show non-hidden rows
+          return filterValue || !row.original.hidden;
+        },
+        enableColumnFilter: true,
+        enableHiding: false,
+        meta: {
+          className: "rt-hidden",
+          headerClassName: "rt-hidden",
+        },
+        cell: () => null,
+        header: () => null,
+        size: 0,
+      }),
+      columnHelper.accessor("user_name", {
+        id: "user_name",
+        header: () => I18n.t("activerecord.attributes.user.user_name"),
+        enableColumnFilter: true,
+      }),
+      columnHelper.accessor("first_name", {
+        header: () => I18n.t("activerecord.attributes.user.first_name"),
+        enableColumnFilter: true,
+      }),
+      columnHelper.accessor("last_name", {
+        header: () => I18n.t("activerecord.attributes.user.last_name"),
+        enableColumnFilter: true,
+      }),
+    ];
+  };
 
   updateShowHidden = event => {
     let showHidden = event.target.checked;
-    let filtered = [];
-    for (let i = 0; i < this.state.filtered.length; i++) {
-      if (this.state.filtered[i].id !== "hidden") {
-        filtered.push(this.state.filtered[i]);
+    let columnFilters = [];
+
+    for (let i = 0; i < this.state.columnFilters.length; i++) {
+      if (this.state.columnFilters[i].id !== "hidden") {
+        columnFilters.push(this.state.columnFilters[i]);
       }
     }
+
     if (!showHidden) {
-      filtered.push({id: "hidden", value: false});
+      columnFilters.push({id: "hidden", value: false});
     }
-    this.setState({filtered, showHidden});
+    this.setState({columnFilters, showHidden});
   };
 
   dataColumns = () => {
-    let columns = [];
-    this.props.assessments.map(data => {
-      columns.push({
-        accessor: `assessment_marks.${data["id"]}.mark`,
-        Header: data["name"],
-        minWidth: 50,
-        className: "number",
-        headerStyle: {textAlign: "right"},
-      });
+    const columnHelper = this.columnHelper;
+    const columns = [];
+
+    this.props.assessments.forEach(data => {
+      columns.push(
+        columnHelper.accessor(`assessment_marks.${data.id}.mark`, {
+          // Custom filter function for numeric columns
+          filterFn: (row, columnId, filterValue) => {
```

Comment:
I don't thinks we should be doing substring matching, but instead exact numeric matching. If the user inputs "1" we should only be looking for scores of "1", not any score that contains a "1".

There are [built-in filter functions](https://tanstack.com/table/latest/docs/guide/column-filtering#filterfns), and `equalsString` may be useful. But if that doesn't work (e.g. if it doesn't handle nulls well) you can stick with a custom-defined function, please just make sure to do exact matching.